File: 28-02-2026/letter_and_number_count.py
import re

def count_letters_and_numbers(s):

    digits = re.findall(r"\d", s)
    letters = re.findall(r"[a-zA-Z]", s)
    dStr, sStr = "number", "letter"

    if len(letters) != 1:
        sStr += "s"
    if len(digits) != 1:
        dStr += "s"

    return f"The string has {len(letters)} {sStr} and {len(digits)} {dStr}."



print(count_letters_and_numbers("helloworld123"))
print(count_letters_and_numbers("A1!"))
print(count_letters_and_numbers("12345"))
print(count_letters_and_numbers("password"))

File: 28-02-2026/letter_and_number_count.js
function countLettersAndNumbers(str) {
  const [allLetters, allNumbers] = [
    str.match(/[a-zA-Z]/gi)?.length || 0,
    str.match(/\d/g)?.length || 0,
  ];

  return `The string has ${allLetters} letter${allLetters !== 1 ? "s" : ""} and ${allNumbers} number${allNumbers !== 1 ? "s" : ""}.`;
}

console.log(countLettersAndNumbers("password")); // "The string has 8 letters and 0 numbers."
console.log(countLettersAndNumbers("12345")); // "The string has 0 letters and 5 numbers."
console.log(countLettersAndNumbers("A1!")); // "The string has 1 letter and 1 number."
console.log(countLettersAndNumbers("Catch 22")); // "The string has 5 letters and 2 numbers."
console.log(countLettersAndNumbers("helloworld123")); // "The string has 10 letters and 3 numbers."
